### ISSUE TITLE
Support "no-cache" header

### ISSUE THREAD

**Issue by gmontard** [plataformatec/faraday-http-cache#81]
Hello,

What I'm trying to achieve is simple, make **API request always check against the server if they haven't changed (using etag) and if they haven't serve the last cached response stored by Faraday HTTP Cache** (using "no-cache" header).

Here a detailed scenario:
- Faraday get a response with a "no-cache" header and an "etag" -> it stores the response in its cache
- Next time we query this resource, faraday ask it again to the server sending the last etag it cached
- If the resource hasn't changed the server respond with a 404 and doesn't send the response over HTTP
- Faraday serve the cached response

It seems this scenario is not implemented as of today, am i right?

Thanks.

PS: For those not familiar with the "no-cache" header it doesn't mean it shouldn't be cached
> “No-cache” shows that returned responses can’t be used for subsequent requests to the same URL before checking if server responses have changed. If a proper ETag (validation token) is present as a result, no-cache incurs a roundtrip in an effort to validate cached responses. Caches can however eliminate downloads if the resources haven’t changed. In other words, web browsers might cache the assets but they have to check on every request if the assets have changed (304 response if nothing has changed). (https://www.keycdn.com/blog/a-guide-to-http-cache-headers/)

**Comment by nitzanav** [comment]
I needed that too, and this is the code I used:

```ruby
class Faraday::OverrideCacheControl < Faraday::Middleware
  def initialize(app, options = {})
    super(app)

    @cache_control = options[:cache_control]
  end

  def call(env)
    response = @app.call(env)
    response.headers['Cache-Control'] = @cache_control

    response
  end
end

class Faraday::CachedClient
  def self.client
    Faraday.new do |builder|
      builder.use :http_cache,
                  store: ActiveSupport::Cache.lookup_store(:file_store, "#{Rails.root}/tmp/cache"),
                  shared_cache: false,
                  logger: Rails.logger,
                  serializer: Marshal
      builder.use Faraday::OverrideCacheControl, cache_control: 'public, max-age=604800'
      builder.adapter Faraday.default_adapter
    end
  end
end

Faraday.default_connection = Faraday::CachedClient.client

Faraday.get('http://path/to/no_cache')
```

There is another solution: https://github.com/ismasan/faraday_throttler

**Comment by lucasmazza** [comment]
> It seems this scenario is not implemented as of today, am i right?

Yeah, we don't have that implemented yet - just the `no-cache` support for requests. 

It shouldn't be too hard though - we can check for that directive on the `Response#fresh?` method (so it always return false for `no-cache` responses) and the middleware should follow the revalidation logic directly.

**Comment by lucasmazza** [comment]
Fix released at `1.3.0` :smile:

**Comment by gmontard** [comment]
:+1: Thanks!
